app/main.py

```diff
@@ -2,5 +2,14 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_memory = {}
+
+    def wrapper(*args) -> dict:
```

Comment:
Consider using a decorator for `wrapper` from `functools` to preserve the wrapped function naming and docstring.